### PR TITLE
sql: support neg for apd

### PR DESCRIPTION
**Pull request by sploiselle**

### CODE REVIEW

File: src/expr/src/scalar/func.rs [hunk 1]
```diff
@@ -1321,6 +1321,16 @@ fn neg_decimal<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(-a.unwrap_decimal())
 }
 
+fn neg_apd<'a>(a: Datum<'a>) -> Datum<'a> {
+    let mut a = a.unwrap_apd();
+    // 0 and NaN can be negated in the underlying library, but must not be
+    // negated in SQL.
+    if !(a.0.is_zero() || a.0.is_nan()) {
+        apd::cx_datum().neg(&mut a.0);
+    }
```
Comment (benesch) [inline src/expr/src/scalar/func.rs:1330]: I'm a little worried about trying to enforce this adhoc. We might want to write a function (`munge_apd`) that gets called after every op on an APD that handles this.

**Review by benesch — APPROVED** [review]

File: src/expr/src/scalar/func.rs (hunk 1 above)
Comment (sploiselle) [reply src/expr/src/scalar/func.rs:1330]: Ah, ok––I had thought of that but it seemed a little wasteful on CPU cycles for things we would need to test for in the first place. If you prefer `munge_apd`, though, glad to fix that up here.

**Comment by benesch** [comment]
No, not necessarily prefer, just saying I’m nervous that this won’t catch
all cases! I’m happy to be proven wrong, but also it’s always ok to prefer
correctness to performance, especially in the absence of a demonstrated
perf issue.

**Review by philip-stoev — APPROVED** [review]
Thank you for the test -- I do not have any cases to add here.

I will be adding unary minus to my randomized workload.

**Comment by sploiselle** [comment]
@benesch Added a `munge_apd` function; lmk what you think

**Review by benesch — APPROVED** [review]
Looks great to me!

**Pull request merged**